### PR TITLE
fix(apprunner): startCommand and environment are ignored in imageConfiguration 

### PR DESCRIPTION
**Pull request by pahud**
This PR addresses the following issues

1. custom environment variables and start commands should be allowed for `imageConfiguration`
2. buildCommand, environment and startCommand should be allowed for `codeConfigurationValues`

Fixes: #16812 

- [x] add tests
- [x] implementation

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apprunner/lib/service.ts [hunk 1]
```diff
@@ -798,25 +809,44 @@ export class Service extends cdk.Resource {
...
   private renderImageRepository(): any {
     const repo = this.source.imageRepository!;
+    this.environment = repo.imageConfiguration?.environment;
     if (repo.imageConfiguration?.port) {
       // convert port from type number to string
       return Object.assign(repo, {
         imageConfiguration: {
           port: repo.imageConfiguration.port.toString(),
+          startCommand: repo.imageConfiguration.startCommand,
+          runtimeEnvironmentVariables: this.renderEnvironmentVariables(),
```
Comment (corymhall) [inline packages/@aws-cdk/aws-apprunner/lib/service.ts:830]: What happens to these values when `port` is not provided?

Comment (pahud) [reply packages/@aws-cdk/aws-apprunner/lib/service.ts]: Thanks for pointing out this stupid mistake. Should all fixed. Please take a look again.  @corymhall

Comment (pahud) [reply packages/@aws-cdk/aws-apprunner/lib/service.ts]: Hey @neilkuan are you able to take a peek at this? 😸

File: packages/@aws-cdk/aws-apprunner/lib/service.ts [hunk 2]
```diff
@@ -798,22 +809,36 @@ export class Service extends cdk.Resource {
...
+      },
+    });
+  }
+
+  private renderEnvironmentVariables(): EnvironmentVariable[] | undefined {
+    if (this.environment) {
+      let env: EnvironmentVariable[] = [];
+      for (const [key, value] of Object.entries(this.environment)) {
+        env.push({ name: key, value: value });
+      }
```
[the remark this answers is not in the document]
Comment (pahud) [reply packages/@aws-cdk/aws-apprunner/lib/service.ts:838]: nice catch!

[the remark this answers is not in the document]
Comment (pahud) [reply packages/@aws-cdk/aws-apprunner/lib/service.ts]: fixed. Can you take a peek again?

**Review by neilkuan — COMMENTED** [review]
LGTM

**Review by corymhall — APPROVED** [review]

**Pull request merged**